widgets/search-box.js

```diff
@@ -1,45 +1,63 @@
 var React = require('react');
 
 var utils = require('../lib/utils.js');
+var headerFooter = require('../decorators/headerFooter');
+var SearchBox = headerFooter(require('../components/SearchBox'));
 
-function searchbox(params) {
-  var SearchBox = require('../components/SearchBox');
-
-  var container = utils.getContainerNode(params.container);
+/**
+ * Instantiate a searchbox
+ * @param  {String|DOMElement} options.container CSS Selector or DOMElement to insert the widget
+ * @param  {String} [options.placeholder='Search here'] Input's placeholder
+ * @param  {Object} [options.cssClasses] CSS classes to add to the wrapping elements: root, input
+ * @param  {String|String[]} [options.cssClasses.root=null]
+ * @param  {String|String[]} [options.cssClasses.input=null]
+ * @param  {Object} [options.templates] Templates to use for the widget
+ * @param  {String|Function} [options.templates.header=''] Header template
+ * @param  {String|Function} [options.templates.footer=''] Footer template
+ * @param  {boolean} [poweredBy=false] Show a powered by Algolia link below the input
+ * @return {Object}
+ */
+function searchbox({
+    container = null,
+    placeholder = 'Search here',
+    cssClasses = {
+      input: null,
+      root: null
+    },
+    templates = {
+      header: '',
+      footer: ''
+    },
+    poweredBy = false
+  }) {
+  var containerNode = utils.getContainerNode(container);
   var isFocused = false;
 
   return {
-    init: function(initialState, helper) {
+    _render(state, helper) {
```

Comment:
Yes I was thinking about defining inside the searchBox function so we can benefit from the closure.

My main argument was not readability but making this function not dependant on `this`. But this is not really a big deal in our case, because we are the users of those function (via instantSearch.js). I'm fine with leaving it in the object :)